### PR TITLE
Add multi GPU training benchmarks

### PR DESCRIPTION
**Pull request by DamianSzwichtenberg**
Benchmarks currently cover only homogeneous cases. All tested configurations show improvement in loss over epochs.

### CODE REVIEW

File: benchmark/multi_gpu/training/training_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
+
+    if args.device == 'xpu':
+        model, optimizer = ipex.optimize(model, optimizer=optimizer)
+
+    train = train_hetero if hetero else train_homo
+
+    if rank == 0:
+        maybe_synchronize(args.device)
```
Comment (akihironitta) [inline benchmark/multi_gpu/training/training_benchmark.py:201]: n00b question: Is it correct that it's synchronised on rank zero only? Should it be on all ranks?

**Review by akihironitta — COMMENTED** [review]
🚀

File: benchmark/multi_gpu/training/training_benchmark.py (hunk 1 above)
Comment (DamianSzwichtenberg) [reply benchmark/multi_gpu/training/training_benchmark.py]: That's a very good question and a good catch. Indeed I should synchronize on all ranks. Thanks 😉 just fixed that.

**Review by JakubPietrakIntel — APPROVED** [review]
LGTM
